src/backend/generator.js

```diff
@@ -542,13 +553,15 @@ exports.createDistDir = function(req, socket, callback) {
         if (err) {
           console.log(err);
           logger.addLog('Error', 'Error occured when calculating hash of event folder', socket, err);
+          callback(null);
           return done(err);
         }
         distHelper.copyServiceWorker(appFolder, hashObj.hash, function(error) {
           if (error) {
             console.log(error);
             logger.addLog('Error', 'Error occured while copying service worker file', socket, err);
-            return done(error);
+            callback(null);
+            return done(err);
```

Comment:
I think this should be `return done(error);`

And the `addLog()` line above should use `error` too!

(By this point in the code, `err` is unset.)

Reply:
Yes, that should be `error` thanks.